### DISCUSSION TITLE
2.0 API reference should use select function instead of query function

### DISCUSSION THREAD

**Discussion by KantiCodes** [sqlalchemy/sqlalchemy#7658] [Ideas]
Hi there!

I am not sure if this intentional **2.0 API** tutorial the examples still use **1.X.X** style - for example [here](https://docs.sqlalchemy.org/en/20/orm/inheritance_loading.html#loading-objects-with-joined-table-inheritance)

As per migration notes from below:
![image](https://user-images.githubusercontent.com/24612001/152158973-6ebd5a7e-9b48-4614-8c9e-e6c9fdb72edf.png)

It might be just me but I found it a bit confusing and only after some time I realized that it's probably ported from 1.X.X tutorial. 

I suspect that it's something that's on your _TODO_ list or its been simply decided that people will figure it out (oops me :D) In any case I just leave it here as a reminder!

Have a wonderful day!

**Comment by zzzeek** [comment]
hey there  -

oh absolutely, we have an enormous job converting all the examples to use new style queries.   if you have any inclination to send PRs for various doc sections, we're all for that.

**Reply by CaselIT** [reply]
let's create an issue with help wanted and pin it

edit: created https://github.com/sqlalchemy/sqlalchemy/issues/7659 and pinned in the issue page

**Reply by KantiCodes** [reply]
Hi both! Thanks so much for fast reply it's a great feeling when maintainers reply within an hour!

This would be my first open-source contribution - challenge accepted!

I will try to update the **loading-inheritance-hierarchies** tutorial. Disclaimer: I am basic-orm SQLAlchemy user and these concepts are fairly new to me (hence me reading the docs today :) )

When reading about option for ```with_polymorphic()``` **1.X.X** lists 3 options

> Basic control of this behavior is provided using the with_polymorphic() function, as well as two variants, the mapper configuration mapper.with_polymorphic in conjunction with the mapper.polymorphic_load option, and the Query -level Query.with_polymorphic() method. 

Considering that ```Query``` was removed and I couldn't find anything on ```select``` do we then have two in **2.0**? 

> Basic control of this behavior is provided using the with_polymorphic() function, as well as the mapper configuration mapper.with_polymorphic in conjunction with the mapper.polymorphic_load option.

**Reply by zzzeek** [reply]
query level with_polymorphic has been removed.   the other two things are still there.   the with_polymorphic() function works with select() or Query().
